### PR TITLE
frost-link test suite

### PR DESCRIPTION
**Pull request by juwara0**
Created frost-link test suite
Cleaned up README and demo app
Refactored frost-link component.js file to remove deprecation for double render and make component more testable.

### CODE REVIEW

File: addon/components/frost-link.js [hunk 1]
```diff
@@ -1,70 +1,59 @@
...
 
+/**
+ * List of valid values to pass into `design` propery
+ * @type {Array} valid `design` values
+ */
 const validDesigns = [
   'info-bar',
   'in-line',
   'inline'
 ]
```
Comment (sglanzer-fire) [inline addon/components/frost-link.js:21]: Are we dropping `design`?  I believe we had discussed that with respect to the button

File: addon/components/frost-link.js [hunk 2]
```diff
@@ -1,70 +1,59 @@
...
 
-  attributeBindings: [
-    'disabled'
-  ],
+  /*
+    The Link component provides and sets default values for:
+    disabled: false - className and attributeBindings
+    tabindex: null - attributeBindings
+    target: null
+   */
```
Comment (sglanzer-fire) [inline addon/components/frost-link.js:50]: Nice 👍

File: addon/components/frost-link.js [hunk 3]
```diff
@@ -138,10 +163,13 @@ export default LinkComponent.extend(PropTypeMixin, {
...
     this._super(...arguments)
 
-    // Turn off the default _invoke on event and use _clickAndInvoke instead
-    let eventName = get(this, 'eventName')
-    this.off(eventName)
-    this.on(eventName, this, this._clickAndInvoke)
+    this._setTarget()
+  },
+
+  click () {
```
Comment (sglanzer-fire) [inline addon/components/frost-link.js:171]: Do we want to set this up with the `EventProxy` mixin?  @srowhani any thoughts?

File: addon/components/frost-link.js [hunk 4]
```diff
@@ -83,19 +71,25 @@ export default LinkComponent.extend(PropTypeMixin, {
       design: '',
       icon: '',
       priority: '',
-      size: '',
-      text: ''
```
Comment (sglanzer-fire) [inline addon/components/frost-link.js:87]: In order to data-drive links I'd like to see if we can figure out how to keep `text` (and make it actually work) - I think we can piggyback on `linkTitle` for this https://github.com/emberjs/ember.js/blob/v2.7.0/packages/ember-htmlbars/lib/components/link-to.js#L796

File: tests/integration/components/frost-link-test.js [hunk 5]
```diff
@@ -9,18 +18,200 @@ describeComponent(
...
+        this.$('.frost-link').text().trim(),
+        'Yields content'
+      ).to.eql('Yielded content')
+    })
+
+    describe('Priority property', function () {
+      it('has primary class set', function () {
+        this.render(hbs`
+          {{#frost-link
+            'test'
```
Comment (sglanzer-fire) [inline tests/integration/components/frost-link-test.js]: Just a preference, but I like to see the positional params inline with the component (I kind of see them as function parameters)

e.g. 

```
{{#frost-link 'test'
  priority='primary'
}}
  Yielded content
{{/frost-link}}
```

**Comment by sglanzer-fire** [comment]
Good stuff @juwara0 - just one note, do we need to add an acceptance test as well to make sure that the link actually links to a location?  Or are we relying on `link-to` for that?

File: addon/components/frost-link.js (hunk 1 above)
Comment (juwara0) [reply addon/components/frost-link.js:21]: Yes, I have captured this in an issue so we can track it.

File: addon/components/frost-link.js (hunk 2 above)
Comment (juwara0) [reply addon/components/frost-link.js:50]: Thanks.

File: addon/components/frost-link.js (hunk 3 above)
Comment (juwara0) [reply addon/components/frost-link.js:171]: I was wondering about that as well. I noticed that we are using the `EventProxy` for `frost-password`. I will open an issue to track moving to `EventProxy` for all of the components that it makes sense to do so.

Comment (sglanzer-fire) [reply addon/components/frost-link.js:171]: Perfect, thanks @juwara0!

File: tests/integration/components/frost-link-test.js (hunk 5 above)
Comment (juwara0) [reply tests/integration/components/frost-link-test.js]: Updated.

**Comment by juwara0** [comment]
Per the acceptance question note. For testing of the link actually linking to a location, we can rely on the link-to tests. That scenario is covered.

**Comment by sglanzer-fire** [comment]
Approved - good stuff

File: addon/components/frost-link.js (hunk 4 above)
Comment (juwara0) [reply addon/components/frost-link.js:87]: Done

**Pull request merged**